mlflow/utils/search_utils.py

```diff
@@ -10,15 +11,23 @@ class SearchFilter(object):
     VALID_METRIC_COMPARATORS = set(['>', '>=', '!=', '=', '<', '<='])
     VALID_PARAM_COMPARATORS = set(['!=', '='])
     VALID_TAG_COMPARATORS = set(['!=', '='])
+    VALID_STRING_ATTRIBUTE_COMPARATORS = set(['!=', '='])
+    VALID_NUMERIC_ATTRIBUTE_COMPARATORS = set(['>', '>=', '!=', '=', '<', '<='])
+    VALID_ATTRIBUTE_KEYS = set(RunInfo.get_attributes())
+    VALID_NUMERIC_ATTRIBUTE_KEYS = set(RunInfo.get_numeric_attributes())
+    VALID_STRING_ATTRIBUTE_KEYS = VALID_ATTRIBUTE_KEYS - VALID_NUMERIC_ATTRIBUTE_KEYS
     _METRIC_IDENTIFIER = "metric"
     _ALTERNATE_METRIC_IDENTIFIERS = set(["metrics"])
     _PARAM_IDENTIFIER = "parameter"
     _ALTERNATE_PARAM_IDENTIFIERS = set(["param", "params"])
     _TAG_IDENTIFIER = "tag"
     _ALTERNATE_TAG_IDENTIFIERS = set(["tags"])
+    _ATTRIBUTE_IDENTIFIER = "attribute"
+    _ALTERNATE_ATTRIBUTE_IDENTIFIERS = set(["attr", "run"])
```

Comment:
micronit: should we allow plural "attributes" here since we allow plural "metrics", "params", "tags"?